release-23.2: catalog/lease: Improve reliability of TestLeaseRenewedAutomatically

spilchen opened the pull request:
Backport 1/1 commits from #134665.

/cc @cockroachdb/release

---

The TestLeaseRenewedAutomatically test is designed to verify that an expired lease is reacquired automatically. However, background processes acquiring leases caused occasional test flakiness. Efforts to limit this interference, such as disabling autostats, were only partially effective, as other background activity still sometimes caused issues. Rather than attempting to track every background process, I modified the test to focus lease checks exclusively on the two tables under test. This improved reliability in my local runs since it was the database descriptor lease, not the leases for the two tables, that was being acquired. This adjustment preserves the test’s intent while enhancing its stability.

Epic: none
Closes #133030
Release note: none
Release justification: low risk test only change

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/134722)
<!-- Reviewable:end -->

rafiss reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134722)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @spilchen)

___
*[`pkg/sql/catalog/lease/lease_test.go` line 1250 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134722#-OBEEaB25o_-yKt_RVtv:-OBEEaB3EdSyhnSRjwY3:biivvc) ([raw file](https://github.com/cockroachdb/cockroach/blob/adda6ae26a0ef64907877b01f0ebebf1145d07d2/pkg/sql/catalog/lease/lease_test.go#L1250)):*
> ```Go
> 	var testAcquisitionBlockCount int32
> 	// Descriptor IDs for the two tables under test
> 	var test1ID int32
> ```

super nit: it may not be worth going through and redoing the backports, but just as a tip for other cases where we use atomics, we could use the `atomic.Int32` type to make the ergonomics slightly easier and makes it more clear that the variables must be accessed atomically.


<!-- Sent from Reviewable.io -->

spilchen committed:
catalog/lease: Improve reliability of TestLeaseRenewedAutomatically

The TestLeaseRenewedAutomatically test is designed to verify that an
expired lease is reacquired automatically. However, background processes
acquiring leases caused occasional test flakiness. Efforts to limit this
interference, such as disabling autostats, were only partially
effective, as other background activity still sometimes caused issues.
Rather than attempting to track every background process, I modified the
test to focus lease checks exclusively on the two tables under test.
This improved reliability in my local runs since it was the database
descriptor lease, not the leases for the two tables, that was being
acquired. This adjustment preserves the test’s intent while enhancing
its stability.

Epic: none
Closes #133030
Release note: none
